### PR TITLE
Add authentication and authorization docs for minimal APIs

### PR DESCRIPTION
**Pull request by captainsafia**
[Preview View](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis/security?view=aspnetcore-7.0&branch=pr-en-us-27356)

### CODE REVIEW

**Review by Rick-Anderson — COMMENTED** [review]
Realize this is draft but though I'd start.
Per[ MS style guide](https://learn.microsoft.com/en-us/style-guide/top-10-tips-style-voice#use-bigger-ideas-fewer-words) ***Shorter is always better***, avoid FT (future tense) and slang.

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddAuthentication().AddJwtBearer();
+
+var app = builder.Build();
+
+app.Run();
+```
+
+By default, the  [`WebApplication`](dotnet/api/microsoft.aspnetcore.builder.webapplication) automatically registers the authentication and authorization middlewares if certain authentication and authorization services are enabled. In the following code sample below, it's not necessary to invoke [`UseAuthentication`](dotnet/api/microsoft.aspnetcore.builder.authappbuilderextensions.useauthentication) or [`UseAuthorization`](/dotnet/api/microsoft.aspnetcore.builder.authorizationappbuilderextensions.useauthorization) to register the middlewares since [`WebApplication`](dotnet/api/microsoft.aspnetcore.builder.webapplication) does this automatically after services are added.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:54]: This will fix the build error
```suggestion
By default, the [`WebApplication`](/dotnet/api/microsoft.aspnetcore.builder.webapplication) automatically registers the authentication and authorization middlewares if certain authentication and authorization services are enabled. In the following code sample below, it's not necessary to invoke [`UseAuthentication`](dotnet/api/microsoft.aspnetcore.builder.authappbuilderextensions.useauthentication) or [`UseAuthorization`](/dotnet/api/microsoft.aspnetcore.builder.authorizationappbuilderextensions.useauthorization) to register the middlewares since [`WebApplication`](dotnet/api/microsoft.aspnetcore.builder.webapplication) does this automatically after services are added.
```

**Comment by captainsafia** [comment]
> Realize this is draft but though I'd start.

Yes, I've been force-pushing aggressively into this but am taking it out of draft mode now.

**Comment by Rick-Anderson** [comment]
https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis/security?view=aspnetcore-7.0&branch=pr-en-us-27356

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+  .AddJwtBearer("LocalAuthIssuer");
+  
+var app = builder.Build();
+
+app.Run();
+```
+
+## Configuring authorization policies in minimal applications
+
+While authentication is used to identify and validate the identity of users against an API, authorization is used to validate and verify access to resources in an API and is faciliated by the `IAuthorizationService` registered by the `AddAuthorization` extension method. In the scenario below, we want to add an `/hello` resource that requires a user to present an `is_admin` claim with a `greetings_api` scope.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:136]: Must fix spelling facilitated if you reject my suggestion. Sentence is way to long for MT (machine translation).

```suggestion
Authentication is used to identify and validate the identity of users against an API. Authorization is used to validate and verify access to resources in an API and is facilitated by the `IAuthorizationService` registered by the `AddAuthorization` extension method. In the following scenario, a `/hello` resource is added that requires a user to present an `is_admin` claim with a `greetings_api` scope.
```

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+  .RequireAuthorization("admin_greetings");
+
+app.Run();
+```
+
+## Using `dotnet user-jwts` to improve development time testing
+
+Throughout this article, we've used an application configured with JWT-bearer based authentication as a sample. JWT bearer-based authentication requires that clients present a token in the request header to validate their identity and claims. Typically, these tokens are issued by a central authority, like an identity server.
+
+However, when developing locally, it is helpful to generate these tokens locally and for testing. To that end, the `dotnet user-jwts` tool is helpful for creating bearer tokens.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:176]:
```suggestion
When developing on the local machine, the [`dotnet user-jwts`](/aspnet/core/security/authentication/jwt-authn) tool can be used to creating bearer tokens.
```

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+```dotnetcli
+dotnet user-jwts create
+```
+
+> [!NOTE]
+> When invoked on a project, the tool will automatically add the authentication options matching the generated token to appsettings.json.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:183]:
```suggestion
> When invoked on a project, the tool automatically adds the authentication options matching the generated token to `appsettings.json`.
```

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+Tokens can be configured with a variety of customizations. For example, to create a token provides the `admin` role and `greetings_api` scope expected by the authorization policy above:
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:185]:
```suggestion
Tokens can be configured with a variety of customizations. For example, to create a token for the `admin` role and `greetings_api` scope expected by the authorization policy in the preceding code:
```

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+dotnet user-jwts create --scope "greetings_api" --role "admin"
+```
+
+The generated token can then be sent as part of the header in the testing tool of choice. For example, with curl:
+
+```dotnetcli
+curl -i -H "Authorization: Bearer {token}" https://localhost:{port}/hello
+```
+
+For more information on the `dotnet user-jwts` tool, read the [complete documentation](/aspnet/core/security/authentication/jwt-authn?).
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:197]:
```suggestion
For more information on the `dotnet user-jwts` tool, read the [complete documentation](/aspnet/core/security/authentication/jwt-authn).
```

**Review by Rick-Anderson — APPROVED** [review]
Feel free to reject most of my suggestions. One spelling error and most of the future tense suggestions should be accepted.

**Comment by Rick-Anderson** [comment]
@captainsafia  as soon as you S&M, I'll hook this up to the TOC

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+author: captainsafia
+description: Learn how to configure authentication and authorization in minimal API apps
+ms.author: safia
+monikerRange: '>= aspnetcore-7.0'
+ms.date: 10/17/2022
+uid: fundamentals/minimal-apis/security
+---
+
+# Authentication and authorization in minimal APIs
+
```
Comment (BrennanConroy) [inline aspnetcore/fundamentals/minimal-apis/security.md:12]: nit: Having nothing between the title and the first header doesn't look very appealing on the docs site.

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+## Key concepts in authentication and authorization
+
+Minimal APIs support all the authentication and authorization options available in ASP.NET and provide some additional functionality to improve the experience for working with authentication. Authentication is the process of determining a user's identity. Authorization is the process of determining whether a user has access to a resource. Both authentication and authorization scenarios share similar implementation semantics in ASP.NET Core. Authentication is handled by the authentication service, [IAuthenticationService](/dotnet/api/microsoft.aspnetcore.authentication.iauthenticationservice), which is used by authentication [middleware](/aspnet/core/fundamentals/middleware). Authorization is handled by the authorization service, [IAuthorizationService](/dotnet/api/microsoft.aspnetcore.authorization.iauthorizationservice), which is used by the authorization middleware.
```
Comment (BrennanConroy) [inline aspnetcore/fundamentals/minimal-apis/security.md:15]: nit: ASP.NET Core > ASP.NET

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+
+builder.Services.AddAuthentication().AddJwtBearer();
+builder.Services.AddAuthorization();
+
+var app = builder.Build();
+
+app.Run();
+```
+
+In some cases, such as controlling middleware order, it's necessary to explicitly register authentication and authorization. In the following sample, the authentication middleware runs _after_ the CORS middleware has run.
```
Comment (BrennanConroy) [inline aspnetcore/fundamentals/minimal-apis/security.md:67]: nit: this concept is also covered in https://learn.microsoft.com/aspnet/core/fundamentals/minimal-apis/middleware?view=aspnetcore-7.0
Can we crosslink or something?

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+```
+
+### Configuring authentication strategy
+
+Authentication strategies typically support a variety of configurations that are loaded in via options. Minimal application support loading options from configuration for the following authentication strategies:
+
+- JWT bearer-based authentication strategies
+- OpenID Connection-based authentication strategies
+
+The ASP.NET framework expects to find these options under the `Authentication:Schemes:{SchemeName}` section in configuration. The `appsettings.json` definition below defines two different schemes, `Bearer` and `LocalAuthIssuer`, with their respective options. The `Authentication:DefaultScheme` option can be used to configure the default authentication strategy that will be used.
```
Comment (BrennanConroy) [inline aspnetcore/fundamentals/minimal-apis/security.md:91]: Crosslink to configuration docs?

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+          "http://localhost:5259"
+        ],
+        "ValidIssuer": "local-auth"
+      }
+    }
+  }
+}
+```
+
+In `Program.cs`, we register two JWT bearer-based authentication strategies: one with the default scheme name ("Bearer") and one with the "LocalAuthIssuer" scheme name.
```
Comment (BrennanConroy) [inline aspnetcore/fundamentals/minimal-apis/security.md:117]: This is a little confusing, we just explained setting a default scheme and used "LocalAuthIssuer" as the default, but now we're claiming the default scheme is "Bearer".

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+Tokens can be configured with a variety of customizations. For example, to create a token provides the `admin` role and `greetings_api` scope expected by the authorization policy above:
+
+```dotnetcli
+dotnet user-jwts create --scope "greetings_api" --role "admin"
+```
+
+The generated token can then be sent as part of the header in the testing tool of choice. For example, with curl:
+
+```dotnetcli
+curl -i -H "Authorization: Bearer {token}" https://localhost:{port}/hello
```
Comment (BrennanConroy) [inline aspnetcore/fundamentals/minimal-apis/security.md:194]: nit: `"Authorization: Bearer <token>"`
I think `<something>` is the more common way to express "replace me"

File: aspnetcore/fundamentals/minimal-apis/security.md (hunk 9 above)
Comment (captainsafia) [reply aspnetcore/fundamentals/minimal-apis/security.md:67]: Nice! I can cross-link. I'll include something here as well since this is new behavior.

File: aspnetcore/fundamentals/minimal-apis/security.md (hunk 11 above)
Comment (captainsafia) [reply aspnetcore/fundamentals/minimal-apis/security.md:117]: Ah good insight -- I'll reword this to make it clear we are overriding the framework default.

File: aspnetcore/fundamentals/minimal-apis/security.md (hunk 12 above)
Comment (captainsafia) [reply aspnetcore/fundamentals/minimal-apis/security.md:193]: `{}` feels like a better options since `<>` might be overloaded with the tag syntax that is used elsewhere in the docs and `{}` follows more nicely with the template string syntax.

@Rick-Anderson What's the standard style here?

Comment (Rick-Anderson) [reply aspnetcore/fundamentals/minimal-apis/security.md:195]: > `{}` feels like a better options since `<>` might be overloaded with the tag syntax that is used elsewhere in the docs and `{}` follows more nicely with the template string syntax.
> 
> @Rick-Anderson What's the standard style here?

I agree `{}` is better and more obvious. We've had a couple of complaints about `<>`. cc @tdykstra

File: aspnetcore/fundamentals/minimal-apis/security.md (hunk 8 above)
Comment (Rick-Anderson) [reply aspnetcore/fundamentals/minimal-apis/security.md]: `@DamianEdwards`  won't be happy dropping Core on ASP.NET Core.

Comment (BrennanConroy) [reply aspnetcore/fundamentals/minimal-apis/security.md]: ?? Greater than, not ->

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+description: Learn how to configure authentication and authorization in minimal API apps
+ms.author: safia
+monikerRange: '>= aspnetcore-7.0'
+ms.date: 10/17/2022
+uid: fundamentals/minimal-apis/security
+---
+
+# Authentication and authorization in minimal APIs
+
+Minimal APIs support all the authentication and authorization options available in ASP.NET and provide some additional functionality to improve the experience working with authentication.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:13]: > ?? Greater than, not ->

```suggestion
Minimal APIs support all the authentication and authorization options available in ASP.NET Core and provide some additional functionality to improve the experience working with authentication.
```

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+
+The authentication service uses registered authentication handlers to complete authentication-related actions. For example, an authentication-related action is authenticating a user or logging out a user. Authentication schemes are names that are used to uniquely identify an authentication handler and its configuration options. Authentication handlers are responsible for implementing the strategies for authentication and generating a user's claims given a particular authentication strategy, such as OAuth or OIDC. The configuration options are unique to the strategy as well and provide the handler with configuration that affects authentication behavior, such as redirect URIs.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:19]:
```suggestion
The authentication service uses registered authentication handlers to complete authentication-related actions. For example, an authentication-related action is authenticating a user or signing out a user. Authentication schemes are names that are used to uniquely identify an authentication handler and its configuration options. Authentication handlers are responsible for implementing the strategies for authentication and generating a user's claims given a particular authentication strategy, such as OAuth or OIDC. The configuration options are unique to the strategy as well and provide the handler with configuration that affects authentication behavior, such as redirect URIs.
```

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+
+There are two strategies for determining user access to resources in the authorization layer:
+
+* Role-based strategies determine a user's access based on the role they are assigned, such as `Administrator` or `User`. For more information on role-based authorization, see [role-based authorization documentation](/aspnet/core/security/authorization/roles).
+* Claim-based strategies determine a user's access based on claims that are issued by a central authority. For more information on claim-based authorization, see [claim-based authorization documentation](/aspnet/core/security/authorization/claims).
+
+In ASP.NET, both strategies are captured into an authorization requirement. The authorization service leverages authorization handlers to determine whether or not a particular user meets the authorization requirements applied onto a resource.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:26]:
```suggestion
In ASP.NET Core, both strategies are captured into an authorization requirement. The authorization service leverages authorization handlers to determine whether or not a particular user meets the authorization requirements applied onto a resource. 
```

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 16]
```diff
@@ -0,0 +1,198 @@
...
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddAuthentication().AddJwtBearer();
+
+var app = builder.Build();
+
+app.Run();
+```
+
+By default, the [`WebApplication`](/dotnet/api/microsoft.aspnetcore.builder.webapplication) automatically registers the authentication and authorization middlewares if certain authentication and authorization services are enabled. In the following code sample below, it's not necessary to invoke [`UseAuthentication`](/dotnet/api/microsoft.aspnetcore.builder.authappbuilderextensions.useauthentication) or [`UseAuthorization`](/dotnet/api/microsoft.aspnetcore.builder.authorizationappbuilderextensions.useauthorization) to register the middlewares beause [`WebApplication`](/dotnet/api/microsoft.aspnetcore.builder.webapplication) does this automatically after `AddAuthentication` or `AddAuthorization` are called.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:56]:
```suggestion
By default, the [`WebApplication`](/dotnet/api/microsoft.aspnetcore.builder.webapplication) automatically registers the authentication and authorization middlewares if certain authentication and authorization services are enabled. In the following code sample, it's not necessary to invoke [`UseAuthentication`](/dotnet/api/microsoft.aspnetcore.builder.authappbuilderextensions.useauthentication) or [`UseAuthorization`](/dotnet/api/microsoft.aspnetcore.builder.authorizationappbuilderextensions.useauthorization) to register the middlewares beause [`WebApplication`](/dotnet/api/microsoft.aspnetcore.builder.webapplication) does this automatically after `AddAuthentication` or `AddAuthorization` are called.
```

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 17]
```diff
@@ -0,0 +1,198 @@
...
+
+builder.Services.AddAuthentication().AddJwtBearer();
+builder.Services.AddAuthorization();
+
+var app = builder.Build();
+
+app.Run();
+```
+
+In some cases, such as controlling middleware order, it's necessary to explicitly register authentication and authorization. In the following sample, the authentication middleware runs _after_ the CORS middleware has run. For more information on middlewares and this automatic behavior, review [the documentation](/aspnet/core/fundamentals/minimal-apis/middleware).
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:69]:
```suggestion
In some cases, such as controlling middleware order, it's necessary to explicitly register authentication and authorization. In the following sample, the authentication middleware runs _after_ the CORS middleware has run. For more information on middlewares and this automatic behavior, see [Middleware in Minimal API apps](/aspnet/core/fundamentals/minimal-apis/middleware).
```

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 18]
```diff
@@ -0,0 +1,198 @@
...
+
+app.Run();
+```
+
+### Configuring authentication strategy
+
+Authentication strategies typically support a variety of configurations that are loaded via options. Minimal app's support loading options from configuration for the following authentication strategies:
+
+- JWT bearer-based authentication strategies
+- OpenID Connection-based authentication strategies
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:91]:
```suggestion
- JWT bearer-based
- OpenID Connection-based
```

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 19]
```diff
@@ -0,0 +1,198 @@
...
+
+The ASP.NET Core framework expects to find these options under the `Authentication:Schemes:{SchemeName}` section in configuration. The `appsettings.json` definition in the following sample defines two different schemes, `Bearer` and `LocalAuthIssuer`, with their respective options. The `Authentication:DefaultScheme` option can be used to configure the default authentication strategy that is used.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:93]:
```suggestion
The ASP.NET Core framework expects to find these options under the `Authentication:Schemes:{SchemeName}` section in configuration. The `appsettings.json` definition in the following sample defines two different schemes, `Bearer` and `LocalAuthIssuer`, with their respective options. The `Authentication:DefaultScheme` option can be used to configure the default authentication strategy that's used.
```

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 20]
```diff
@@ -0,0 +1,198 @@
...
+          "http://localhost:5259"
+        ],
+        "ValidIssuer": "local-auth"
+      }
+    }
+  }
+}
+```
+
+In `Program.cs`, we register two JWT bearer-based authentication strategies: one with the the "Bearer" scheme name and one with the "LocalAuthIssuer" scheme name. "Bearer" is the typical default scheme in JWT-bearer based enabled applications, but the default scheme can be overridden by setting the `DefaultScheme` property as in the example above.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:119]:
```suggestion
In `Program.cs`, we register two JWT bearer-based authentication strategies: one with the the "Bearer" scheme name and one with the "LocalAuthIssuer" scheme name. "Bearer" is the typical default scheme in JWT-bearer based enabled applications, but the default scheme can be overridden by setting the `DefaultScheme` property as in the preceding example.
```

File: aspnetcore/fundamentals/minimal-apis/security.md (hunk 12 above)
Comment (tdykstra) [reply aspnetcore/fundamentals/minimal-apis/security.md:195]: The [contributor guide calls for angle brackets](https://review.learn.microsoft.com/en-us/help/contribute/text-formatting-guidelines?branch=main#placeholders), adding that curly braces are OK only for languages like HTML where angle brackets have a special meaning.

>We discourage curly braces { } for use as syntactical placeholders. Readers can confuse curly brace placeholders with the same notation used in:
>
>Replaceable text
> * Format strings
> * String interpolation
> * Text templates
> * Similar programming constructs
>
>Consider using curly braces instead of angle brackets when angle brackets are normally used in the syntax of a language, such as in XML or HTML.

Comment (Rick-Anderson) [reply aspnetcore/fundamentals/minimal-apis/security.md:195]: > The [contributor guide calls for angle brackets](https://review.learn.microsoft.com/en-us/help/contribute/text-formatting-guidelines?branch=main#placeholders), adding that curly braces are OK only for languages like HTML where angle brackets have a special meaning.
> 

The `contributor guide`  needs to be updated to include C#

Comment (captainsafia) [reply aspnetcore/fundamentals/minimal-apis/security.md:195]: > The contributor guide needs to be updated to include C#

Would the C# guide be updated to recommend `{}`. IMO, that makes more sense given the proximity to template strings in C#.

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 21]
```diff
@@ -0,0 +1,198 @@
...
+var app = builder.Build();
+
+app.Run();
+```
+
+Typically, a specific authentication strategy is used. In the following sample, the app is configured with support for JWT bearer-based authentication.
+
+```csharp
+using Microsoft.AspNetCore.Authentication.Cookie;
+
```
Comment (JamesNK) [inline aspnetcore/fundamentals/minimal-apis/security.md:46]:
```suggestion
```
I don't see anything that uses this namespace

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 22]
```diff
@@ -0,0 +1,198 @@
...
+In ASP.NET Core, both strategies are captured into an authorization requirement. The authorization service leverages authorization handlers to determine whether or not a particular user meets the authorization requirements applied onto a resource. 
+
+## Enabling authentication in minimal applications
+
+To enable authentication, call [`AddAuthentication`](/dotnet/api/microsoft.extensions.dependencyinjection.authenticationservicecollectionextensions.addauthentication) to register the required authentication services on the app's service provider.
+
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddAuthentication();
```
Comment (JamesNK) [inline aspnetcore/fundamentals/minimal-apis/security.md:35]: nit: Add code to an external file, then import and highlight this line. Makes it clear to the reader on what to focus on

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 23]
```diff
@@ -0,0 +1,198 @@
...
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddAuthentication().AddJwtBearer();
+
+var app = builder.Build();
+
+app.Run();
+```
+
+By default, the [`WebApplication`](/dotnet/api/microsoft.aspnetcore.builder.webapplication) automatically registers the authentication and authorization middlewares if certain authentication and authorization services are enabled. In the following sample, it's not necessary to invoke [`UseAuthentication`](/dotnet/api/microsoft.aspnetcore.builder.authappbuilderextensions.useauthentication) or [`UseAuthorization`](/dotnet/api/microsoft.aspnetcore.builder.authorizationappbuilderextensions.useauthorization) to register the middlewares because [`WebApplication`](/dotnet/api/microsoft.aspnetcore.builder.webapplication) does this automatically after `AddAuthentication` or `AddAuthorization` are called.
```
Comment (JamesNK) [inline aspnetcore/fundamentals/minimal-apis/security.md:56]: API links should always use xref. Is that right @Rick-Anderson?

I've found https://xrefgen.azurewebsites.net/ makes xref links easy to add.

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 24]
```diff
@@ -0,0 +1,198 @@
+---
+title: Configuring authentication for minimal APIs
+author: captainsafia
+description: Learn how to configure authentication and authorization in minimal API apps
+ms.author: safia
+monikerRange: '>= aspnetcore-7.0'
```
Comment (JamesNK) [inline aspnetcore/fundamentals/minimal-apis/security.md:6]: What is specific to .NET 7? Is it worth adding some conditional tags so this doc is visible with .NET 6?

File: aspnetcore/fundamentals/minimal-apis/security.md (hunk 23 above)
Comment (Rick-Anderson) [reply aspnetcore/fundamentals/minimal-apis/security.md:56]: > I've found https://xrefgen.azurewebsites.net/ makes xref links easy to add.

https://rs-xrefs.azurewebsites.net/ is actually much better but it ran out of money and won't work again until Nov 2nd. I've deleted a few resource groups so that shouldn't happen again.

> API links should always use xref. Is that right @Rick-Anderson? Or is the plan to clean them up later?
> 

We prefer them but I accept either. @tdykstra  knows the official rules.

File: aspnetcore/fundamentals/minimal-apis/security.md (hunk 22 above)
Comment (Rick-Anderson) [reply aspnetcore/fundamentals/minimal-apis/security.md:35]: > nit: Add code to an external file, then import and highlight this line. Makes it clear to the reader on what to focus on
> 
> Repeat for other samples.

Or create an issue with these instructions and I'll do that.

File: aspnetcore/fundamentals/minimal-apis/security.md (hunk 12 above)
Comment (Rick-Anderson) [reply aspnetcore/fundamentals/minimal-apis/security.md:195]: C# is a language where `<>` is part of the language.

File: aspnetcore/fundamentals/minimal-apis/security.md (hunk 24 above)
Comment (captainsafia) [reply aspnetcore/fundamentals/minimal-apis/security.md:6]: The flow documented here (like being able to read options from config and the user-jwts tool) are .NET 7-specific.

File: aspnetcore/fundamentals/minimal-apis/security.md (hunk 23 above)
Comment (captainsafia) [reply aspnetcore/fundamentals/minimal-apis/security.md:56]: The Markdown-style links are easier to produce on the editing side so that's why I tend to rely on them.

File: aspnetcore/fundamentals/minimal-apis/security.md (hunk 22 above)
Comment (captainsafia) [reply aspnetcore/fundamentals/minimal-apis/security.md:35]: Yep, it's a little bit easier to review the code when it is contextualized to the content being mentioned. Once it's finalized, it makes sense to move to a separate file as a follow-up.

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 25]
```diff
@@ -0,0 +1,196 @@
...
+## Configuring authorization policies in minimal applications
+
+Authentication is used to identify and validate the identity of users against an API. Authorization is used to validate and verify access to resources in an API and is facilitated by the `IAuthorizationService` registered by the `AddAuthorization` extension method. In the following scenario, a `/hello` resource is added that requires a user to present an `is_admin` claim with a `greetings_api` scope.
+
+Configuring authorization requirements on a resource is a two-step process that requires:
+
+1. Configuring the authorization requirements in a policy globally.
+2. Applying individual policies to resources.
+
+In the code sample below, the  `AddAuthorizationBuilder` is invoked which:
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:142]:
```suggestion
In the following code, the  `AddAuthorizationBuilder` is invoked which:
```

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 26]
```diff
@@ -0,0 +1,196 @@
...
+
+- Adds authorization-related services to the DI container.
+- Returns an `AuthorizationBuilder` that can be used to directly register authentication policies.
+
+The code creates a new authorization policy, named `policy_greetings`, that encapsulate two authorization requirements:
+
+- A role-based requirement that the user fall under the `admin` role.
+- A claim-based requirement that the user provide a `greetings_api` scope.
+
+Finally, the `admin_greetings` policy is provided as a required policy to the `/hello` endpoint.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:152]:
```suggestion
The `admin_greetings` policy is provided as a required policy to the `/hello` endpoint.
```

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 27]
```diff
@@ -0,0 +1,196 @@
...
+
+app.MapGet("/hello", () => "Hello world!")
+  .RequireAuthorization("admin_greetings");
+
+app.Run();
+```
+
+## Using `dotnet user-jwts` to improve development time testing
+
+Throughout this article, we've used an application configured with JWT-bearer based authentication as a sample. JWT bearer-based authentication requires that clients present a token in the request header to validate their identity and claims. Typically, these tokens are issued by a central authority, like an identity server.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:173]:
```suggestion
Throughout this article, an app configured with JWT-bearer based authentication is used. JWT bearer-based authentication requires that clients present a token in the request header to validate their identity and claims. Typically, these tokens are issued by a central authority, such as an identity server.
```

File: aspnetcore/fundamentals/minimal-apis/security.md [hunk 28]
```diff
@@ -0,0 +1,196 @@
...
+
+When developing on the local machine, the [`dotnet user-jwts`](/aspnet/core/security/authentication/jwt-authn) tool can be used to creating bearer tokens.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/security.md:175]:
```suggestion
When developing on the local machine, the [`dotnet user-jwts`](/aspnet/core/security/authentication/jwt-authn) tool can be used to create bearer tokens.
```

**Review by Rick-Anderson — APPROVED** [review]
With suggestions applied.
